File: src/main/java/br/com/zup/academy/dominio/repository/LivroRepository.java
package br.com.zup.academy.dominio.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import br.com.zup.academy.dominio.modelo.Livro;

@Repository
public interface LivroRepository extends JpaRepository<Livro, Long>{

	@Query("SELECT l FROM Livro l JOIN FETCH l.autor WHERE l.id = :id")
	Optional<Livro> consultarPorId(@Param("id") Long id);
}

File: casa-do-codigo-grpc/src/main/kotlin/br/com/zup/dominio/repository/LivroRepository.kt
package br.com.zup.dominio.repository

import br.com.zup.dominio.modelo.Livro
import io.micronaut.data.annotation.Query
import io.micronaut.data.annotation.Repository
import io.micronaut.data.jpa.repository.JpaRepository
import java.util.*

@Repository
interface LivroRepository: JpaRepository<Livro, Long> {

    @Query("SELECT l FROM Livro l JOIN FETCH l.autor WHERE l.id = :id")
    fun consultarComAutor(id: Long): Optional<Livro>
}
